### PR TITLE
Fixes export SA Parameters

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

**Comment by msevestre** [comment]
@Yuri05 I am going to merge this so that I can update OSPSuite-R and have abdullah update his SA code. 
Please review

File: src/OSPSuite.Infrastructure.Import/Services/PKParameterSensitivitiesImporter.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            return Array.Empty<PKParameterSensitivity>();
+         }
+         catch (Exception e)
+         {
+            logger.AddError(e.FullMessage());
+            return Array.Empty<PKParameterSensitivity>();
+         }
+      }
+
+      private IReadOnlyList<PKParameterSensitivity> pkParameterSensitivitiesFrom(string fileFullPath, char delimiter)
```
Comment (msevestre) [inline src/OSPSuite.Infrastructure.Import/Services/PKParameterSensitivitiesImporter.cs]: Very simple import. Same logic as for importing results

File: src/OSPSuite.Infrastructure.Import/Services/SensitivityAnalysisRunResultsImportTask.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Core.Services;
+using OSPSuite.Utility.Events;
+using OSPSuite.Utility.Extensions;
+
+namespace OSPSuite.Infrastructure.Import.Services
+{
+   public interface ISensitivityAnalysisRunResultsImportTask
+   {
+      Task<SensitivityAnalysisRunResultsImport> ImportResults(IModelCoreSimulation simulation, IReadOnlyCollection<string> files, CancellationToken cancellationToken, bool showImportProgress = true);
```
Comment (msevestre) [inline src/OSPSuite.Infrastructure.Import/Services/SensitivityAnalysisRunResultsImportTask.cs]: Can import SARunResults from multiple files. This was missing for the SA workflow

File: src/OSPSuite.Infrastructure.Import/Services/SensitivityAnalysisRunResultsImportTask.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+      private void addPKParameterSensitivitiesFromSingleFile(SensitivityAnalysisRunResultsImport sensitivityAnalysisRunResultsImport, PKParameterSensitivitiesImport pkParameterSensitivitiesImport)
+      {
+         var sensitivityAnalysisRunResult = sensitivityAnalysisRunResultsImport.SensitivityAnalysisRunResult;
+         sensitivityAnalysisRunResultsImport.PKParameterSensitivitiesImportFiles.Add(pkParameterSensitivitiesImport.PKParameterSensitivitiesImportFile);
+
+         foreach (var pkParameterSensitivity in pkParameterSensitivitiesImport.PKParameterSensitivities)
+         {
+            if (sensitivityAnalysisRunResult.HasPKParameterSensitivityWithId(pkParameterSensitivity.Id))
+               sensitivityAnalysisRunResultsImport.AddError(Error.DuplicatedPKParameterSensitivityFor(pkParameterSensitivity.Id));
```
Comment (msevestre) [inline src/OSPSuite.Infrastructure.Import/Services/SensitivityAnalysisRunResultsImportTask.cs]: We error here if there is two entries for the smae QuantityPath, PKParameter, Parameter combo. This should never happen

File: src/OSPSuite.Infrastructure.Import/Services/SimulationResultsImporter.cs [hunk 4]
```diff
@@ -33,44 +32,36 @@ public SimulationResultsImportTask(IEntitiesInSimulationRetriever quantitiesRetr
...
+               tasks.Remove(firstFinishedTask);
+
+               // Await the completed task. 
+               allImportedResults.Add(await firstFinishedTask);
+               progressUpdater.IncrementProgress();
             }
-         }
-         finally
-         {
-            _allQuantities.Clear();
```
Comment (msevestre) [inline src/OSPSuite.Infrastructure.Import/Services/SimulationResultsImporter.cs]: no need for this guy here global. It's used once, and created where it's used

File: src/OSPSuite.R/Services/ContainerTask.cs [hunk 5]
```diff
@@ -18,6 +18,7 @@ public interface IContainerTask
       IContainer[] AllContainersMatching(IModelCoreSimulation simulation, string path);
       IQuantity[] AllQuantitiesMatching(IModelCoreSimulation simulation, string path);
       IMoleculeAmount[] AllMoleculesMatching(IModelCoreSimulation simulation, string path);
+      IParameter[] AllSensitivityAnalysisParametersMatching(ISimulation simulation, string path);
```
Comment (msevestre) [inline src/OSPSuite.R/Services/ContainerTask.cs]: I have added this one here that returns all SA parameters with a  wild card
Not useful for reporting engine but probably neat when checking only the influence of parameters withing a specifc organ for example

File: src/OSPSuite.R/Services/SensitivityAnalysisTask.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      /// <returns>An array of parameter path</returns>
+      string[] PotentialVariableParameterPathsFor(ISimulation simulation);
+
+      /// <summary>
+      ///    Exports sensitivity analysis run results to the csv file with path <paramref name="csvFile" />
+      /// </summary>
+      /// <param name="sensitivityAnalysisRunResult">Sensitivity analysis run result to export</param>
+      /// <param name="simulation">Simulation used</param>
+      /// <param name="csvFile">Full path of csv file where results should be exported</param>
+      void ExportResultsToCSV(SensitivityAnalysisRunResult sensitivityAnalysisRunResult, ISimulation simulation, string csvFile);
```
Comment (msevestre) [inline src/OSPSuite.R/Services/SensitivityAnalysisTask.cs]: All potential parameters, export and import nwo available in R

File: src/OSPSuite.Infrastructure.Import/Services/PKParameterSensitivitiesImportFile.cs [hunk 7]
```diff
@@ -0,0 +1,7 @@
+﻿namespace OSPSuite.Infrastructure.Import.Services
+{
+   public class PKParameterSensitivitiesImportFile : ImportLogger
+   {
+      
+   }
+}
```
Comment (Yuri05) [inline src/OSPSuite.Infrastructure.Import/Services/PKParameterSensitivitiesImportFile.cs]: ? What is this class for?

File: src/OSPSuite.R/Services/ContainerTask.cs (hunk 5 above)
Comment (Yuri05) [reply src/OSPSuite.R/Services/ContainerTask.cs]: Will this return ALL parameters matching the path, or only meaningful (used in model, non-categorical etc.)?

File: src/OSPSuite.Infrastructure.Import/Services/PKParameterSensitivitiesImportFile.cs (hunk 7 above)
Comment (msevestre) [reply src/OSPSuite.Infrastructure.Import/Services/PKParameterSensitivitiesImportFile.cs]: It is used just to be consistent with all other files import where we have a dedicated class. I can remove it if you prefer.

File: src/OSPSuite.R/Services/ContainerTask.cs (hunk 5 above)
Comment (msevestre) [reply src/OSPSuite.R/Services/ContainerTask.cs]: yes. I am going to rename it to AllParametersForSensitivityAnalysisMatching. This makes more sense I believe

Comment (msevestre) [reply src/OSPSuite.R/Services/ContainerTask.cs]: >Will this return ALL parameters matching the pat 

No that would not make sense. This is AllParametersMatching

**Pull request merged**